### PR TITLE
Source data criteria as models

### PR DESCRIPTION
**Pull request by daco101**
This PR updates the converter to use cqm-models for the source data criteria, and updates some fixtures to work with the new cqm-models branch. It gets rid of data_critria.

Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

- [x] Update Gemfile after https://github.com/projecttacoma/cqm-models/pull/91 is merged

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1931
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If applicable, the library version number in `cqm-converter.gemspec` has been updated
- [x] Update cqm-execution test fixtures using update_cqm_execution_fixtures.sh script from server-scripts repo  https://github.com/projecttacoma/cqm-execution/pull/35

**Bonnie Reviewer:**

Name: @hossenlopp 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**~Cypress~ Bonnie Reviewer 2:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: spec/cqm/converter/bonnie_measure_spec.rb [hunk 1]
```diff
@@ -48,6 +46,17 @@
     expect(population_set.populations.IPP.hqmf_id).to eq('6CD39B4B-16E6-4FD4-9241-527F9F4A48D0')
     expect(population_set.populations.DENOM.statement_name).to eq('Denominator')
     expect(population_set.populations.DENOM.hqmf_id).to eq('B735F4BD-DFAE-45E3-BAA9-E09BDF731B8E')
+
+    # check source data criteria
```
Comment (losborne) [inline spec/cqm/converter/bonnie_measure_spec.rb]: This block should be pulled out into a function as it's repeated below a few times.  Maybe name it `check_source_data_criteria` so the comment can be removed as well

File: lib/cqm/converter/bonnie_measure.rb [hunk 2]
```diff
@@ -155,6 +154,19 @@ def shallow_copy(bonnie_measure)
   cqm_measure
 end
 
+def convert_source_data_criteria(bonnie_measure)
+  @map_definition_and_status_to_model ||= JSON.parse(File.read(File.join(File.dirname(__FILE__), 'map_definition_and_status_to_model.json')))
+  converted_source_data_criteria = bonnie_measure.source_data_criteria.map do |_, sdc|
+    key = "#{sdc['definition']}::#{sdc['status']}"
+    model_name = @map_definition_and_status_to_model[key]['model_name']
```
Comment (hossenlopp) [inline lib/cqm/converter/bonnie_measure.rb]: Suggest checking if something at `key` exists before continuing. If it doesn't exist then raise an error detailing the type that was not found. This would allow our batch script to log failed conversions.
